src/init.cpp

```diff
@@ -1966,7 +1970,20 @@ bool StartIndexBackgroundSync(NodeContext& node)
         }
     }
 
+    std::shared_ptr<ThreadPool> thread_pool;
+    if (node.args->IsArgSet("-indexworkers")) {
+        int index_workers = node.args->GetIntArg("-indexworkers", INDEX_WORKERS_COUNT);
+        if (index_workers < 0 || index_workers > 100) return InitError(_("Invalid -indexworkers arg"));
+
+        thread_pool = std::make_shared<ThreadPool>();
+        thread_pool->Start(index_workers);
```

Comment:
Since the constructor and `Start` are always called in succession in this patch set, you could make `ThreadPool` more RAII styled if the `Start` method were removed and instead placed in the constructor. Could also make sense to do the same with the destructor and `Stop`.